src/main/pages/che-7/hands-on-guidance/proc_creating-a-workspace-by-importing-source-code-of-your-project.adoc

```diff
@@ -1,27 +1,111 @@
 [id="creating-a-workspace-by-importing-source-code-of-your-project_{context}"]
-= Creating a workspace by importing source code of your project
+= Creating a workspace by importing the source code of your project
 
-This paragraph is the procedure module introduction: a short description of the procedure.
+This section is about how to create a new workspace to edit an existing codebase.
+
+There are 2 ways to do that *before* starting a workspace:
+
+* xref:#configure-devfile[Select a stack from the User Dashboard, then change the devfile to include your project]
+* xref:#configuring-a-che-workspace-using-a-devfile_end-user-tasks[Add a devfile to a git repository and start the workspace using chectl or a factory]
+
+
+There are also 3 ways to do that *after* starting a workspace:
+
+* xref:#importing-from-the-user-dashboard-into-an-existing-workspace[Import from the User Dashboard into an existing workspace]
+* xref:#importing-to-a-running-workspace-with-git-clone-command[Import to a running workspace with git clone command]
+* xref:#importing-to-a-running-workspace-with-git-clone-in-a-terminal[Import to a running workspace with git clone in a terminal]
 
```

Comment:
* Import a workspace from the *Dashboard* into an existing workspace. For details, see xref:#importing-from-the-user-dashboard-into-an-existing-workspace.
* Import to a running workspace using the `git clone` command. For details, see xref:#importing-to-a-running-workspace-with-git-clone-in-a-terminal.
* Import to a running workspace using the `git clone` command in a terminal. For details, see xref:#importing-to-a-running-workspace-with-git-clone-in-a-terminal.

Reply:
ugh, ok.